src/app/meditation/meditation.component.ts

```diff
@@ -359,6 +366,48 @@ export class MeditationComponent {
     this.appState.set('isMeditating', false);
   }
 
+  /**
+   * Determines the percentage of the reached goal of commitment.
+   */
+  reached(commitment) {
```

Comment:
This is mostly copied from the commitment component, right? Could you please move it to its own service?